configs/seL4Config.cmake

```diff
@@ -184,71 +187,98 @@ unset(KernelArmMachFeatureModifiers CACHE)
 
-# Now enshrine all the common variables in the config
-config_set(KernelArmCortexA7 ARM_CORTEX_A7 "${KernelArmCortexA7}")
-config_set(KernelArmCortexA8 ARM_CORTEX_A8 "${KernelArmCortexA8}")
-config_set(KernelArmCortexA9 ARM_CORTEX_A9 "${KernelArmCortexA9}")
-config_set(KernelArmCortexA15 ARM_CORTEX_A15 "${KernelArmCortexA15}")
-config_set(KernelArmCortexA35 ARM_CORTEX_A35 "${KernelArmCortexA35}")
-config_set(KernelArmCortexA53 ARM_CORTEX_A53 "${KernelArmCortexA53}")
-config_set(KernelArmCortexA57 ARM_CORTEX_A57 "${KernelArmCortexA57}")
-config_set(KernelArmCortexA72 ARM_CORTEX_A72 "${KernelArmCortexA72}")
-config_set(KernelArm1136JF_S ARM1136JF_S "${KernelArm1136JF_S}")
-config_set(KernelArchArmV6 ARCH_ARM_V6 "${KernelArchArmV6}")
-config_set(KernelArchArmV7a ARCH_ARM_V7A "${KernelArchArmV7a}")
-config_set(KernelArchArmV7ve ARCH_ARM_V7VE "${KernelArchArmV7ve}")
-config_set(KernelArchArmV8a ARCH_ARM_V8A "${KernelArchArmV8a}")
-config_set(KernelArmSMMU ARM_SMMU "${KernelArmSMMU}")
-config_set(KernelAArch64SErrorIgnore AARCH64_SERROR_IGNORE "${KernelAArch64SErrorIgnore}")
+#-------------------------------------------------------------------------------
+# Process platform and architecture specific configuration. Enshrine all the
+# common variables in the config.
+#-------------------------------------------------------------------------------
+
 set(KernelPlatformSupportsMCS "${KernelPlatformSupportsMCS}" CACHE INTERNAL "" FORCE)
 
-# Check for v7ve before v7a as v7ve is a superset and we want to set the
-# actual armv to that, but leave armv7a config enabled for anything that
-# checks directly against it
-if(KernelArchArmV7ve)
-    set(KernelArmArmV "armv7ve" CACHE INTERNAL "")
-elseif(KernelArchArmV7a)
-    set(KernelArmArmV "armv7-a" CACHE INTERNAL "")
-elseif(KernelArchArmV8a)
-    set(KernelArmArmV "armv8-a" CACHE INTERNAL "")
-elseif(KernelArchArmV6)
-    set(KernelArmArmV "armv6" CACHE INTERNAL "")
-endif()
-if(KernelArmCortexA7)
-    set(KernelArmCPU "cortex-a7" CACHE INTERNAL "")
-elseif(KernelArmCortexA8)
-    set(KernelArmCPU "cortex-a8" CACHE INTERNAL "")
-elseif(KernelArmCortexA9)
-    set(KernelArmCPU "cortex-a9" CACHE INTERNAL "")
-elseif(KernelArmCortexA15)
-    set(KernelArmCPU "cortex-a15" CACHE INTERNAL "")
-elseif(KernelArmCortexA35)
-    set(KernelArmCPU "cortex-a35" CACHE INTERNAL "")
-elseif(KernelArmCortexA53)
-    set(KernelArmCPU "cortex-a53" CACHE INTERNAL "")
-elseif(KernelArmCortexA57)
-    set(KernelArmCPU "cortex-a57" CACHE INTERNAL "")
-elseif(KernelArmCortexA72)
-    set(KernelArmCPU "cortex-a72" CACHE INTERNAL "")
-elseif(KernelArm1136JF_S)
-    set(KernelArmCPU "arm1136jf-s" CACHE INTERNAL "")
-endif()
 if(KernelArchARM)
+
+    # Check ARM Cortex-A cores setup. While different Cortex-A version can be
+    # enabled, 'KernelArmCPU' can only be set to one. So we take the first one
+    # from the list, the current ordering ins simply ascending without any
+    # deeper meanding
+    set(first_match "")
```

Comment:
In my opinion this new rewrite of the above code isn't easier to read/maintain/understand because it's got more complex control flow than what it's replacing.  Are you looking at trying to provide error messages for if the platform/config.cmake files make incorrect CPU/ARCH settings? Another way that we could achieve this is by inverting the logic, and requiring the platform files to set the string variants of KernelArmCPU and KernelArmArmV and then handling the boolean options the same way as with KernalSel4Arch.  

Ultimately, it would be better to set the CPU and March from the device tree.  I looked at doing this, but currently hardware_gen.py takes KernelPaddrUserTop as an argument, and KernelPaddrUserTop is set based on the ArmCPU value.  So it would require a bit more effort to implement.